src/database.ts

```diff
@@ -523,13 +525,18 @@ class Database extends common.GrpcServiceObject {
       sessionCount: count,
     };
 
+    const headers = this.resourceHeader_;
+    if ((this.instance.parent as Spanner).routeToLeaderEnabled) {
```

Comment:
Could we maybe refactor these blocks a little bit so that the if-statement also goes in a generic method. So that would probably mean:

1. Add a method like `getSpanner` to all classes that need to be able to add these headers.
2. Modify the `addLeaderAwareRoutingHeader` so it takes a `Spanner` instance as an argument and includes this if-statement

Reply:
Added the method getSpanner in each class.